ttnn/cpp/ttnn/operations/reduction/generic/device/multi_core_h/reduce_op_multi_core_h.cpp

```diff
@@ -53,7 +55,7 @@ operation::ProgramWithCallbacks reduce_multi_core_h(
         tt::tt_metal::split_work_to_cores(compute_with_storage_grid_size, num_cols);
```

Comment:
In case of nd sharding, we should distribute work over sharded cores, no all cores. 
```suggestion
        tt::tt_metal::split_work_to_cores(compute_with_storage_grid_size, num_cols);
        
        if (use_nd_sharding) {
            std::tie(num_cores, all_cores, core_group_1, core_group_2, num_cols_per_core_group_1, num_cols_per_core_group_2) = tt::tt_metal::split_work_to_cores(a.nd_shard_spec().value().grid.bounding_box().grid_size(), num_cols);
        }
```

Reply:
Will this divide the cores in a way where core_group_1 has the cores with num_dev_pages_per_core() shards and core_group_2 has the cores with num_dev_pages_per_core()-1 shards, and ordered in a way such that the first core in core_group_1 has the first shard, the second core the second shard, etc.?